orc8r/cloud/go/mproto/mproto_test.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mproto_test
+
+import (
+	"bytes"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"magma/orc8r/cloud/go/mproto"
+	"magma/orc8r/cloud/go/mproto/protos"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	testDataObjs = []protos.TestData{
+		{Key: "12345", Value: 10},
+		{Key: "23456", Value: 15},
+		{Key: "34567", Value: 20},
+		{Key: "45678", Value: 25},
+		{Key: "56789", Value: 30},
+		{Key: "67890", Value: 35},
+	}
+	dataObjCount = len(testDataObjs)
+
+	testdataDir    = "testdata"
+	goldenFilepath = filepath.Join(testdataDir, "determinisic-digest.golden")
+)
```

Comment:
I don't like this style of initializing test variables as package-level static variables. Maybe for the filepath vars it makes sense, but for the actual data objects, either put them directly in the relevant fns, or place them within a "getter" fn